test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -65,4 +65,67 @@ test.describe( 'Links', () => {
 			},
 		] );
 	} );
+
+	test( 'can update the url of an existing link', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is WordPress' );
+		// Select "WordPress".
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'primary+k' );
+		await page.keyboard.type( 'w.org' );
+
+		await page
+			.locator( '.block-editor-link-control' )
+			.getByRole( 'button', { name: 'Save' } )
+			.click();
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is <a href="http://w.org">WordPress</a>',
+				},
+			},
+		] );
+
+		// Move caret back into the link.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+
+		// Edit link.
+		await pageUtils.pressKeys( 'primary+k' );
+		await pageUtils.pressKeys( 'primary+a' );
+		await page.keyboard.type( 'wordpress.org' );
+
+		// Update the link.
+		await page.keyboard.press( 'Enter' );
+
+		// Navigate back to the popover.
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+
+		// Navigate back to inputs to verify appears as changed.
+		await pageUtils.pressKeys( 'primary+k' );
+		const urlInputValue = await page
+			.getByPlaceholder( 'Search or type url' )
```

Comment:
Nit: I wonder if we could use `getByRole` here? I think either is fine though 👍 .